src/Data/Num/Linear.hs

```diff
@@ -208,38 +211,143 @@ instance (MultIdentity a) => Monoid (Product a) where
 instance (AddIdentity a) => Monoid (Sum a) where
   mempty = Sum zero
 
+{- ORMOLU_DISABLE -}
```

Comment:
I took the liberty of re-ordering these by instance rather than by class because I believe it's easier on the eyes. Ormolu is disabled in this part of the code because it would add a newline between each `deriving` line. I hope you agree. If not, we can always change it back.

Reply:
Disabling Ormolu is a little unorthodox, but I must admit that in this case, it looks like it's worth it.